recipes/vulkan-loader/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+            self.requires("xorg/system")
+        if self.options.get_safe("with_wsi_wayland"):
+            self.requires("wayland/1.18.0")
+
+    def package_id(self):
+        if self.settings.compiler == "Visual Studio":
+            self.info.settings.compiler.runtime = str(self.info.settings.compiler.runtime).replace("MD", "MT")
+
+    def build_requirements(self):
+        if self.options.get_safe("with_wsi_xcb") or self.options.get_safe("with_wsi_xlib") or \
+           self.options.get_safe("with_wsi_wayland") or self.options.get_safe("with_wsi_directfb"):
+            self.build_requires("pkgconf/1.7.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(glob.glob("Vulkan-Loader-*")[0], self._source_subfolder)
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "FindVulkanHeaders.cmake"),
+                              "HINTS ${VULKAN_HEADERS_INSTALL_DIR}/share/vulkan/registry",
+                              "HINTS ${VULKAN_HEADERS_INSTALL_DIR}/res/vulkan/registry")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["VULKAN_HEADERS_INSTALL_DIR"] = self.deps_cpp_info["vulkan-headers"].rootpath
+        self._cmake.definitions["BUILD_TESTS"] = False
+        self._cmake.definitions["USE_CCACHE"] = False
+        if self.settings.os == "Linux":
+            self._cmake.definitions["BUILD_WSI_XCB_SUPPORT"] = self.options.with_wsi_xcb
+            self._cmake.definitions["BUILD_WSI_XLIB_SUPPORT"] = self.options.with_wsi_xlib
+            self._cmake.definitions["BUILD_WSI_WAYLAND_SUPPORT"] = self.options.with_wsi_wayland
+            self._cmake.definitions["BUILD_WSI_DIRECTFB_SUPPORT"] = self.options.with_wsi_directfb
+        if self.settings.os == "Windows":
+            self._cmake.definitions["ENABLE_WIN10_ONECORE"] = False
+        if tools.is_apple_os(self.settings.os):
+            self._cmake.definitions["BUILD_STATIC_LOADER"] = not self.options.shared
+        self._cmake.definitions["BUILD_LOADER"] = True
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        if self.deps_cpp_info["vulkan-headers"].version != self.version:
+            raise ConanInvalidConfiguration("vulkan-loader must be built with the same version than vulkan-headers.")
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "loader"))
+
+    def package_info(self):
+        if self.deps_cpp_info["vulkan-headers"].version != self.version:
```

Comment:
Can this check be moved to `validate`?

Reply:
I don't want to prevent to consume vulkan-loader with another version of vulkan-header, see commit message https://github.com/conan-io/conan-center-index/pull/4308/commits/824e22bf87417e693775b7864b42d8cfb3c228cb